nexus/db-queries/src/db/datastore/region.rs

```diff
@@ -301,68 +301,26 @@ impl DataStore {
                 let region_ids = region_ids.clone();
                 async move {
-                    use db::schema::crucible_dataset::dsl as dataset_dsl;
-                    use db::schema::region::dsl as region_dsl;
+                    use db::schema::region::dsl as dsl;
 
-                    // Remove the regions, collecting datasets they're from.
-                    let datasets = diesel::delete(region_dsl::region)
-                        .filter(region_dsl::id.eq_any(region_ids))
-                        .returning(region_dsl::dataset_id)
-                        .get_results_async::<Uuid>(&conn).await?;
+                    // Remove the regions
+                    diesel::delete(dsl::region)
+                        .filter(dsl::id.eq_any(region_ids))
+                        .execute_async(&conn)
+                        .await?;
 
                     // Update datasets to which the regions belonged.
-                    for dataset in datasets {
-                        let dataset_total_occupied_size: Option<
-                            diesel::pg::data_types::PgNumeric,
-                        > = region_dsl::region
-                            .filter(region_dsl::dataset_id.eq(dataset))
-                            .select(diesel::dsl::sum(
-                                region_dsl::block_size
-                                    * region_dsl::blocks_per_extent
-                                    * region_dsl::extent_count,
-                            ))
-                            .nullable()
-                            .get_result_async(&conn).await?;
-
-                        let dataset_total_occupied_size: i64 = if let Some(
-                            dataset_total_occupied_size,
-                        ) =
-                            dataset_total_occupied_size
-                        {
-                            let dataset_total_occupied_size: db::model::ByteCount =
-                                dataset_total_occupied_size.try_into().map_err(
-                                    |e: anyhow::Error| {
-                                        err.bail(RegionDeleteError::NumericError(
-                                            e.to_string(),
-                                        ))
-                                    },
-                                )?;
-
-                            dataset_total_occupied_size.into()
-                        } else {
-                            0
-                        };
-
-                        diesel::update(dataset_dsl::crucible_dataset)
-                            .filter(dataset_dsl::id.eq(dataset))
-                            .set(
-                                dataset_dsl::size_used
-                                    .eq(dataset_total_occupied_size),
-                            )
-                            .execute_async(&conn).await?;
-                    }
+                    // XXX put this file somewhere else
```

Comment:
As you say, we should probably fix this before merging

Reply:
yeah, I thought about this, and ended up embedding it as a string: 4834bd9dd. it's going to have to change independent of that update CTE anyway in the future.